ext/mysql2/client.c

```diff
@@ -880,7 +880,7 @@ static VALUE rb_mysql_client_server_info(VALUE self) {
  *
  * Return the file descriptor number for this client.
  */
-static VALUE rb_mysql_client_socket(VALUE self) {
+static VALUE rb_mysql_client_socket(RB_MYSQL_UNUSED VALUE self) {
```

Comment:
It's a hint to the compiler that the argument won't be used - I think it is only to suppress the warning - I don't know if there is an assembly output or optimization difference (I would be interested to find out! but not the hard way ;))